beacon-chain/blockchain/service_test.go

```diff
@@ -215,9 +216,10 @@ func TestRunningChainService(t *testing.T) {
 
 	chainService.incomingBlockChan <- blockNoParent
 	chainService.incomingBlockChan <- block
+	<-blockChan
 	chainService.cancel()
 	exitRoutine <- true
-	testutil.WaitForLog(t, hook, "Chain service context closed, exiting goroutine")
```

Comment:
This blocking assertion call seems somewhat dangerous. In this case, blocking on `exitRoutine` ensures that it's not necessary anyways.